File: test/fixtures/sample.py
def greet(name):
    return "Hello, " + name


def validate(email):
    if len(email) > 0:
        return True
    return False


def process_user(name, email):
    greeting = greet(name)
    print(greeting)
    if validate(email):
        save_user(name, email)


def save_user(name, email):
    print(f"Saving user: {name}, {email}")


def main():
    process_user("Alice", "alice@test.com")


if __name__ == "__main__":
    main()

File: test/fixtures/sample.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

char* greet(char* name) {
    static char buffer[100];
    sprintf(buffer, "Hello, %s", name);
    return buffer;
}

bool validate(char* email) {
    if (strlen(email) > 0) {
        return true;
    }
    return false;
}

void processUser(char* name, char* email) {
    char* greeting = greet(name);
    printf("%s\n", greeting);
    if (validate(email)) {
        saveUser(name, email);
    }
}

void saveUser(char* name, char* email) {
    printf("Saving user: %s, %s\n", name, email);
}

int main() {
    processUser("Alice", "alice@test.com");
    return 0;
}
